src/fr-device/index.ts

```diff
@@ -0,0 +1,241 @@
+      console.warn('Cannot collect browser metadata. navigator is not defined.');
+      return {};
+    }
+    return this.reduceToObject(this.config.browserProps, navigator);
+  }
+
+  getBrowserPluginsNames(): string {
+    if (!(navigator && navigator.plugins)) {
+      console.warn('Cannot collect browser plugin information. navigator.plugins is not defined.');
+      return '';
+    }
+    return this.reduceToString(Object.keys(navigator.plugins), navigator.plugins);
+  }
+
+  getDeviceName(): string {
+    if (!navigator) {
+      console.warn('Cannot collect device name. navigator is not defined.');
+      return '';
+    }
+    const userAgent = navigator.userAgent;
+    const platform = navigator.platform;
+
+    switch (true) {
+      case this.config.devicePlatforms.mac.indexOf(platform) !== -1:
+        return 'Mac (Browser)';
+      case this.config.devicePlatforms.ios.indexOf(platform) !== -1:
+        return `${platform} (Browser)`;
+      case this.config.devicePlatforms.windows.indexOf(platform) !== -1:
+        return 'Windows (Browser)';
+      case /Android/.test(platform) || /Android/.test(userAgent):
+        return 'Android (Browser)';
+      case /CrOS/.test(userAgent) || /Chromebook/.test(userAgent):
+        return 'Chrome OS (Browser)';
+      case /Linux/.test(platform):
+        return 'Linux (Browser)';
+      default:
+        return `${platform || 'Unknown'} (Browser)`;
+    }
+  }
+
+  getDisplayMeta(): { [key: string]: any } {
+    if (!screen) {
+      console.warn('Cannot collect screen information. screen is not defined.');
+    }
+    return this.reduceToObject(this.config.displayProps, screen);
+  }
+
+  getHardwareMeta(): { [key: string]: string } {
+    if (!navigator) {
+      console.warn('Cannot collect OS metadata. Navigator is not defined.');
+      return {};
+    }
+    return this.reduceToObject(this.config.hardwareProps, navigator);
+  }
+
+  getIdentifier(): string {
+    if (!(window.crypto && window.crypto.getRandomValues)) {
+      console.warn('Cannot generate profile ID. Crypto and/or getRandomValues is not supported.');
+      return '';
+    }
+    if (!localStorage) {
+      console.warn('Cannot store profile ID. localStorage is not supported.');
+      return '';
+    }
+    let id = localStorage.getItem('profile-id');
+    if (!id) {
+      id = window.crypto.getRandomValues(new Uint32Array(3)).join('-');
+      localStorage.setItem('profile-id', id);
+    }
+    return id;
+  }
+
+  getInstalledFonts(): string {
+    const canvas = document.createElement('canvas');
+    if (!canvas) {
+      console.warn('Cannot collect font data. Browser does not support canvas element');
+      return '';
+    }
+    const context = canvas.getContext && canvas.getContext('2d');
+
+    if (!context) {
+      console.warn('Cannot collect font data. Browser does not support 2d canvas context');
+      return '';
+    }
+    const text = 'abcdefghi0123456789';
+    context.font = '72px Comic Sans';
```

Comment:
🤭

Reply:
Ha ha ha, I actually discovered a good use of Comic Sans :D 

Ref: https://remysharp.com/2008/07/08/how-to-detect-if-a-font-is-installed-only-using-javascript/